test-app/resources/todos/post.js

```diff
@@ -1,3 +1,7 @@
+if (this.todoId) {
+  this.id = this.todoId;
```

Comment:
It's not necessary to add `todoId` to the resource. You can just access it by `body.todoId`.